runtime/common/src/main/kotlin/kotlinx/serialization/SerialFormat.kt

```diff
@@ -1,23 +1,29 @@
 package kotlinx.serialization
 
-import kotlinx.serialization.context.*
+import kotlinx.serialization.module.SerialModule
+import kotlinx.serialization.module.plus
 import kotlinx.serialization.internal.HexConverter
 
+internal const val InstallDeprecationText = "Install mutates format instance, pass module in constructor"
+
 interface SerialFormat {
+    @Deprecated(InstallDeprecationText)
     fun install(module: SerialModule)
 
-    val context: SerialContext
+    val context: SerialModule
 }
 
-abstract class AbstractSerialFormat: SerialFormat {
-    protected val mutableContext: MutableSerialContext = MutableSerialContextImpl()
+abstract class AbstractSerialFormat(context: SerialModule): SerialFormat {
+
+    // this could be a constructor val after `install` would be deleted
+    final override var context: SerialModule = context
```

Comment:
This is mostly not for ABI but to maintain source compatibility for at least one release before removing deprecated methods. If `context` is a `val`, it would be impossible for `install` to modify it. Since `install` already in 0.10.0 release, I suggest to deprecate it with a warning for now and then remove in next release or closer to 1.0